Mathlib/Data/List/Shortlex.lean

```diff
@@ -0,0 +1,285 @@
+/-
+Copyright (c) 2024 Hannah Fechtner. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Hannah Fechtner
+-/
+
+import Mathlib.Data.List.Lex
+import Mathlib.Tactic.Linarith
+
+/-!
+# Shortlex ordering of lists.
+
+Given a relation `r` on `α`, the shortlex order on `List α` is defined by `L < M` iff
+* `L.length < M.length`
+* `L.length = M.length` and `L < M` under the lexicographic ordering over `r` on lists
+
+## Main results
+
+We show that if `r` is well-founded, so too is the shortlex order over `r`
+
+## See also
+
+Related files are:
+* `Mathlib.Data.List.Lex`: Lexicographic order on `List α`.
```

Comment:
Can you refer to these by their filenames, with `/`s, instead? I don't think docgen really knows how to parse this style of link.

Reply:
done! if what I've done is correct (merely replacing `.` with `/`), I can also make a small PR updating this in the `List.Lex` file